### PR TITLE
(perf) Remove "interesting headers" to fasten request processing (replaced with `removeHeadersExcept`)

### PR DESCRIPTION
**Pull request by mathieucarbou**
All headers are already parsed: interesting headers allowed to cleanup parsed headers to only keep interesting ones during request processing.

Headers are anyway all cleaned after request processing.

This removal helps reduce flash size and heap usage (on vector less).

### CODE REVIEW

File: src/ESPAsyncWebServer.h [hunk 1]
```diff
@@ -408,6 +409,55 @@ class AsyncWebServerRequest {
...
+    size_t getHeaderNames(std::vector<const char*>& names) const {
+      names.clear();
+      const size_t size = _headers.size();
+      names.reserve(size);
+      for (const auto& h : _headers) {
+        names.push_back(h.name().c_str());
+      }
+      return size;
+    }
+    const std::list<AsyncWebHeader>& getHeaders() { return _headers; }
```
Comment (vortigont) [inline src/ESPAsyncWebServer.h:421]: This method must be cons also?  `const std::list<AsyncWebHeader>& getHeaders() const { return _headers; }`

**Review by vortigont — COMMENTED** [review]
one small remark so far

**Comment by mathieucarbou** [comment]
I will merge this one today: this one is not as important as the middleware one and is quite simple.

**Comment by mathieucarbou** [comment]
And in case you see something, we'll PR a fix.

**Pull request merged**